### DISCUSSION TITLE
Multiple apps bot in the same Docker container

### DISCUSSION THREAD

**Discussion by mkagit** [jakobdylanc/llmcord.py#65] [General]
Hi!
Is it possible to set multiple bot apps but using only one Docker container?
I want to run multiple models in my server but need to avoid deploying one container per app.

**Comment by jakobdylanc** [comment]
Currently it isn't designed to do that, but good point.

You could accomplish this yourself pretty easily with the proper Dockerfile / docker-compose.yaml modifications.

Currently I'm not sure how I would add this to llmcord while keeping things simple. Open to suggestions.

**Reply by mkagit** [reply]
I tried doing the following test: same Discord user has two separate applications. I'm running two separate py files inside the same container (llmcord1 and llmcord2), each one loading its own config1.yaml and config2.yaml file using different models but using the same Groq API key. It seems to be working; both applications are invited and functioning within the same server. However, I’m having an issue with duplication. When one responds to the other, instead of just one response, there are two responses, the first one respond again. I mean, both apps are working, the duplication occours only when triggering to respond one each other.


```
2024-11-05 16:57:58.592 | 2024-11-05 19:57:58,592 INFO: HTTP Request: POST https://api.groq.com/openai/v1/chat/completions "HTTP/1.1 200 OK" |  
-- | -- | --
  |   | 2024-11-05 16:57:58.369 | 2024-11-05 19:57:58,369 INFO: HTTP Request: POST https://api.groq.com/openai/v1/chat/completions "HTTP/1.1 200 OK" |  
  |   | 2024-11-05 16:57:58.309 | Concorda <@124867303165> |  
  |   | 2024-11-05 16:57:58.309 | 2024-11-05 19:57:58,309 INFO: Message received (user ID: 70720923760, attachments: 0, conversation length: 4): |  
  |   | 2024-11-05 16:57:57.989 | Concorda <@124867303165> |  
  |   | 2024-11-05 16:57:57.989 | 2024-11-05 19:57:57,988 INFO: Message received (user ID: 70720923760, attachments: 0, conversation length: 3): |  
  |   | 2024-11-05 16:57:25.791 | 2024-11-05 19:57:25,790 INFO: HTTP Request: POST https://api.groq.com/openai/v1/chat/completions "HTTP/1.1 200 OK" |  
  |   | 2024-11-05 16:57:25.505 | Quem foi Einstein? <@130342645963>

```

**Reply by jakobdylanc** [reply]
What do you mean exactly by respond to each other?

llmcord always rejects incoming messages from bots, so 2 llmcord bots shouldn't be able to trigger each other, unless you changed something.

**Reply by mkagit** [reply]
something like this:

@groq1 APP who is Donald Trump?
Donald Trump is an American businessman, television personality, and politician who served as the 45th President of the United States from 2017 to 2021. He was born on June 14, 1946, in Queens, New York, to Mary Anne MacLeod and Frederick Christ Trump.

< @groq2 APP and how many Trump children from the latest marriage?

@groq1 APP 
Donald Trump has one child, a son named Barron William Trump, born on March 20, 2006.
@groq2 APP
Donald Trump has one child, a son named Barron William Trump, born on March 20, 2006.

I've asked only from @groq2 about the @groq1 response and got 2 responses, one from @groq1 and one from @groq2. I was expecting to receive only the response from the @groq2.

**Reply by jakobdylanc** [reply]
Ah so you're replying to groq1's message? In that case it's supposed to respond. That's how you continue conversations with llmcord.

And since you @mentioned groq2, it responds as well. So this sounds like normal behavior to me.

If you don't want the bot to respond, you can toggle this to `@ OFF` when replying to it's message:
![image](https://github.com/user-attachments/assets/c01f8837-5566-4631-8804-95366eda9ff9)

**Reply by mkagit** [reply]
Yeah, I've replied to groq1 response mentioning @groq2 then I got responses from both. My mind was to ask only @groq2 about groq1's response
